README.md

```diff
@@ -186,6 +124,69 @@ print("Updated params:", m.get_params())
 Module.get_compiled_artifact(m).save("/tmp/some_file.vmfb")
 ```
 
+## What is a JAX program?
+
+The challenge with talking about extracting a JAX program is that in all
+generality, a JAX program is bounded only by what can run in the host Python
+interpreter. What we are looking for is a simpler definition that allows a
+useful set of standalone programs to be constructed and that meshes well with
+the programming model employed by typical JAX developers.
+
+The components that are the most interesting towards this end are:
+
+* `jax.jit` functions: stateless functions mapping inputs to outputs, natively
+  represented as JAXPR IR and universally convertible to MHLO IR.
+* Multi-dimensional arrays very similar to numpy arrays, with extensions to
+  enable tracing and distribution.
+* Trees of arrays and primitives (a structured generalization of nested dicts,
+  lists and tuples).
+* "AbstractValues" mirroring all concrete data types, facilitating symbolic
+  tracing and descriptions of program elements.
+
+There are of course many more details than this, but these are the components
+we will assemble.
+
+## What is an IREE program?
+
+An IREE program is:
+
+* Defined by an MLIR module containing:
+  * A name, which can be used for cross-referencing between modules. All modules
+    loaded into a single runtime instance must have a unique name.
+  * Zero or more globals. Globals can contain any scalar or multi-dimensional
+    array of any supported type (IREE has an open type system and supports
+    more than this but these types are relevant to JAX). Globals can be:
+      * Loaded, yielding their current value
+      * Stored, updating the value
+      * Mutable or immutable
+      * Initialized, either via a constant or via special initializer functions
+        that can do whatever they want at module load time
+  * One or more public functions, each of which accepts and returns scalars and
+    buffers representationing multi-dimensional arrays. IREE functions can
+    do a lot of things, but relevant to JAX, they can include:
+      * A graph of MHLO operations (the native IR of JAX jitted functions)
+      * Operations and types from
+        [IREE's input dialect](https://github.com/google/iree/blob/main/llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/Input/InputOps.td)
+      * Operations from low level MLIR dialects (for programs coming from a high
+        level IR like MHLO, these are typically not used, but they provide
+        the generality needed for more advanced cases such as implementing
+        full language compilers):
+        * MLIR [arith](https://mlir.llvm.org/docs/Dialects/ArithmeticOps/) dialect
+        * MLIR [linalg](https://mlir.llvm.org/docs/Dialects/Linalg/) dialect
+        * MLIR [math](https://mlir.llvm.org/docs/Dialects/MathOps/) dialect
+        * MLIR [std](https://mlir.llvm.org/docs/Dialects/Standard/) dialect (being phased out)
```

Comment:
I'm curious why this is being phased out?

Reply:
Originally it was the only dialect of substance. We've been executing sporadically for years to disaggregate it into seperate dialects for each distinct set of things (arith, math, tensor, memref, and others all used to be part of std). Now there are just some odds and ends left in std.